libbpf-cargo/src/lib.rs

```diff
@@ -88,7 +89,7 @@ mod test;
 #[derive(Error, Debug)]
 pub enum Error {
     #[error("Error building BPF object file: {0}")]
-    Build(String),
+    Build(#[from] anyhow::Error),
```

Comment:
I don't believe this should have a `#[from]` annotation. There is nothing about this variant that is tied to anyhow's `Error`. What happens if we were to switch the `Generate` variant over to using anyhow's `Error` as well? Now we are suddenly implicitly converting anyhow errors into `Build`, right? That seems wrong. `#[from]` is meant for cases where there is a clear relationship between a unique error type and the thing stored in the variant.

Furthermore, I believe it should have the `#[source]` annotation and omit `{0}` in the error description. Otherwise, that will just cause duplicate information from what I understand, depending on how formatting is done.

Reply:
Yes, `#[from]` should be removed. we can use `anyhow::Error` directly without call `into()`. 

`Generate` can also use `anyhow::Error` if you prefer, I will do.

I have changed `Build` and `Generate` to `#[error(transparent)] to forward `Build, Generate` to underlying anyhow::Error

Reply:
`#[error("...")]` only generate Display trait not Debug, `Error building BPF object file:` show when you print like `{}, e`, but context and backtrace only show when you `{:?}, e`

Reply:
Thanks. I am not sure `#[error(transparent)]` is what we want, though. We still want to convey what went wrong (building or generating), otherwise why do we keep this error type to begin with?

So I'd think we want to keep printing the "Error generating skeleton" error string but should leave out the `{0}` part (at least if we use the `#[source]` annotation as well, which would be my preference).